### ISSUE TITLE
Add rule to check for one space between selector and open brace

### ISSUE THREAD

**Issue by umherirrender** [wikimedia/stylelint-config-wikimedia#57]
The missing space in the following css is not detected with the stylelint-config-wikimedia version 0.4.1
```
.articleplaceholder-identifierlist td{
	vertical-align: top;
	padding: 0.25em;
}
```

Please add a rule to check for one space between the td and {

**Comment by ntwb** [comment]
That would be the[block-opening-brace-space-before/](https://stylelint.io/user-guide/rules/block-opening-brace-space-before/)

**Comment by jdforrester** [comment]
Unfortunately, this is because `block-opening-brace-space-before` doesn't let us configure it the way we want.

We have it set to `always-single-line`, which allows `a { color: pink; }` and rejects `a{ color: pink; }` (good), but allows `a{↵color: pink; }` (bad, and the issue here).

However, setting it to `always` broke things, however, because it rejects `a↵{ color: pink; }` (which is a common format for multi-selector blocks). The other values (`never`, `never-single-line`, `always-multi-line`, and `never-multi-line`) also allow behaviours we don't want or prohibit ones we do.

I guess we want to ask to add yet another config to upstream (which is already too confusing) to force this – "single-line-space-or-multi-line-new-line" or something even more Baroque.

**Comment by Volker-E** [comment]
Something must have changed since August 2017. ;)
When setting stylelint to `"block-opening-brace-space-before": "always"` it correctly flags [this](https://github.com/wikimedia/mediawiki-extensions-GlobalPreferences/blob/master/resources/ext.GlobalPreferences.global-nojs.css#L70) as issue:
```lang=css
div.mw-globalprefs-global-check div.mw-input,
div.mw-htmlform-field-HTMLUsersMultiselectField div.mw-input
{
```

I've then tested with 
- MediaWiki core
- VisualEditor
- MinervaNeue
- OOUI
- Vector

and it found a few mistakes and one nasty one in MinervaNeue.
But it doesn't seem to induce any problems (any more?)
So preparing a patch for changing this…
